clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetsForLeaderEpochClient.java

```diff
@@ -61,67 +77,76 @@ protected OffsetForEpochResult handleResponse(
             Map<TopicPartition, SubscriptionState.FetchPosition> requestData,
             OffsetsForLeaderEpochResponse response) {
 
+        Set<TopicPartition> missingPartitions = new HashSet<>(requestData.keySet());
```

Comment:
Wonder if it might be simpler to initialize `partitionsToRetry` from the request key set.

Reply:
The downside of relying on `partitionsToRetry` is that we can't differentiate the partitions to be retried due to an error from the missing partitions in the response. It may not be that important but we can't preserve the above log otherwise.
```
logger().warn("Missing partition {} from response, retrying.", topicPartition);
```

Reply:
No strong opinion. It's an unlikely case anyway, so I'm not sure it calls for special treatment.

Reply:
Yeah, you're right. Let me remove it.